src/Core/Module/ModuleManager.php

```diff
@@ -156,12 +156,12 @@ public function uninstall(string $name, bool $deleteFiles = false): bool
         $module = $this->moduleRepository->getModule($name);
         $uninstalled = $module->onUninstall();
 
-        if ($deleteFiles && $path = $this->moduleRepository->getModulePath($name)) {
-            $this->filesystem->remove($path);
-        }
-
         $this->dispatch(ModuleManagementEvent::UNINSTALL, $module);
 
+        if ($deleteFiles) {
+            return $uninstalled && $this->delete($name);
```

Comment:
Shouldn't this be noted as a BC Break? In the case where `$this->delete()` doesn't return true, the result of the `uninstall` function won't be the same before and after this PR.

(Which is fine, BC breaks are allowed, this is PS 9)